doc/en/requirements.txt

```diff
@@ -4,10 +4,9 @@ sphinx-removed-in>=0.2.0
 sphinx>=7
 sphinxcontrib-trio
 sphinxcontrib-svg2pdfconverter
-# Pin packaging because it no longer handles 'latest' version, which
-# is the version that is assigned to the docs.
-# See https://github.com/pytest-dev/pytest/pull/10578#issuecomment-1348249045.
-packaging
 furo
 sphinxcontrib-towncrier
+# Pin towncrier temporarily due to incompatibility with sphinxcontrib-towncrier:
+# https://github.com/sphinx-contrib/sphinxcontrib-towncrier/issues/92
+towncrier!=24.7.0,!=24.7.1
```

Comment:
May I suggest moving this to a `broken-dep-constraints.txt` and wiring that via `-c broken-dep-constraints.txt` in this file?

Reply:
Sure, can you elaborate on why though?

My initial reaction is that perhaps this is a bit unnecessary given we have few dependencies and even less "broken" constraints, but perhaps there is some other benefit that I'm not seeing?

Reply:
It's good to separate direct dependencies from transitive ones. And this is the first step towards that. The direct dependencies of the docs build venv are things that we invoke/import directly, whereas anything else isn't. We can tell the dependency resolver to exclude some versions from consideration under some circumstances w/o mixing them into the list of the direct deps. The direct deps list shouldn't really have any version limitations, except maybe lower bounds in case when we use a feature that appeared in a certain version.

Reply:
> It's good to separate direct dependencies from transitive ones.

This is a good point, haven't thought of that. 

Will do. 👍